lib/data/datasource_impl/contact/phonebook_contact_datasource_impl.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:fluffychat/data/datasource/phonebook_datasouce.dart';
+import 'package:fluffychat/domain/model/contact/contact.dart';
+import 'package:flutter_contacts/flutter_contacts.dart' hide Contact;
+
+class PhonebookContactDatasourceImpl implements PhonebookContactDatasource {
+  @override
+  Future<List<Contact>> fetchContacts() async {
+    final phonebookContacts =
+        await FlutterContacts.getContacts(withProperties: true);
+    return phonebookContacts
+        .expand(
+          (contact) => contact.emails.isEmpty && contact.phones.isEmpty
+              ? [Contact(displayName: contact.displayName)]
```

Comment:
I think we should ignore these contacts. no email, no phone :)))

Reply:
done